tests/unit/s2n_tls_prf_test.c

```diff
@@ -136,5 +137,86 @@ int main(int argc, char **argv)
+                S2N_DEFAULT_TEST_PRIVATE_KEY));
+
+        struct s2n_config *config = s2n_config_new();
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, tls12_chain_and_key));
+        EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(config));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "20170210"));
+
+        struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(client_conn);
+
+        struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER);
+        EXPECT_NOT_NULL(server_conn);
+
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+
+        DEFER_CLEANUP(struct s2n_stuffer client_to_server = { 0 }, s2n_stuffer_free);
+        DEFER_CLEANUP(struct s2n_stuffer server_to_client = { 0 }, s2n_stuffer_free);
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&client_to_server, 0));
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&server_to_client, 0));
+        EXPECT_SUCCESS(s2n_connection_set_io_stuffers(&server_to_client, &client_to_server, client_conn));
+        EXPECT_SUCCESS(s2n_connection_set_io_stuffers(&client_to_server, &server_to_client, server_conn));
+
+        EXPECT_OK(s2n_negotiate_test_server_and_client_until_message(server_conn, client_conn, CLIENT_KEY));
+
+        /* Client writes Client Key Exchange message */
+        EXPECT_SUCCESS(s2n_handshake_write_io(client_conn));
+
+        uint8_t data[SHA256_DIGEST_LENGTH] = { 0 };
+        struct s2n_blob digest_for_ems = { 0 };
+        EXPECT_SUCCESS(s2n_blob_init(&digest_for_ems, data, sizeof(data)));
+
+        /* Here we fiddle with the io_stuffers to get the Client Key transcript */
+        DEFER_CLEANUP(struct s2n_stuffer client_key_message = { 0 }, s2n_stuffer_free);
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&client_key_message, 0));
+        EXPECT_SUCCESS(s2n_stuffer_skip_read(&client_to_server, S2N_TLS_RECORD_HEADER_LENGTH));
+        uint8_t client_key_message_length = s2n_stuffer_data_available(&client_to_server);
+        EXPECT_SUCCESS(s2n_stuffer_copy(&client_to_server, &client_key_message, client_key_message_length));
+        client_key_message.blob.size = client_key_message_length;
+
+        EXPECT_OK(s2n_retrieve_digest_for_ems(server_conn, &client_key_message.blob, &digest_for_ems));
+
+        /* Server reads Client Key Exchange message */
+        EXPECT_SUCCESS(s2n_stuffer_rewind_read(&client_to_server, S2N_TLS_RECORD_HEADER_LENGTH + client_key_message_length));
+        EXPECT_SUCCESS(s2n_handshake_read_io(server_conn));
+
+        /* Calculate the digest message after the Server read the Client Key message */
+        uint8_t server_digest[SHA256_DIGEST_LENGTH] = { 0 };
+        POSIX_GUARD(s2n_hash_copy(&server_conn->hash_workspace, &server_conn->handshake.sha256));
```

Comment:
This copy is probably unnecessary, since you're not going to continue the handshake. And should you be hardcoding sha256 here?